modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java

```diff
@@ -199,7 +213,84 @@ public List<DeployedService> listDeployedServicesOfIsv(Category category, Csp cs
         String namespace = userServiceHelper.getCurrentUserManageNamespace();
         query.setNamespace(namespace);
         List<ServiceDeploymentEntity> deployServices = serviceDeploymentStorage.listServices(query);
-        return deployServices.stream().map(EntityTransUtils::convertToDeployedService).toList();
+        return setServiceConfigurationForDeployedService(deployServices);
+    }
+
+    private void setServiceConfigurationDetailsForDeployedService(DeployedService details,
+            ServiceDeploymentEntity serviceDeployment) {
+        ServiceTemplateEntity serviceTemplate = serviceTemplateStorage
+                .getServiceTemplateById(serviceDeployment.getServiceTemplateId());
+        if (Objects.nonNull(serviceTemplate)) {
+            ServiceConfigurationManage serviceConfigurationManage =
+                    serviceTemplate.getOcl().getServiceConfigurationManage();
+            if (Objects.nonNull(serviceConfigurationManage)) {
+                List<ServiceConfigurationParameter> configurationParameters =
+                        serviceConfigurationManage.getConfigurationParameters();
+
+                Map<String, Object> configuration =
+                        Objects.nonNull(details.getServiceConfigurationDetails())
+                                ? details.getServiceConfigurationDetails().getConfiguration()
+                                : null;
+
+                ServiceConfigurationDetails serviceConfigurationDetails =
+                        getServiceConfigurationDetails(configurationParameters,
+                                configuration,
+                                serviceTemplate.getLastModifiedTime());
+                details.setServiceConfigurationDetails(serviceConfigurationDetails);
+            }
+        }
+    }
+
+    private ServiceConfigurationDetails getServiceConfigurationDetails(
```

Comment:
rename this to 'mergeConfigurationParametersFromTemplate'
Also add a method doc -> Method to dynamically merge and generate the current configuration of a service in case the service configuration parameters have been updated in the template by the ISV after a service was deloyed.